model/flow/executionResult.go

```diff
@@ -23,3 +22,13 @@ func (er ExecutionResult) ID() Identifier {
 func (er ExecutionResult) Checksum() Identifier {
 	return MakeID(er)
 }
+
+// FinalStateCommit gets the final state of the result
+// if the number of chunks are 0 returns empty statecommit
+func (er ExecutionResult) FinalStateCommit() StateCommitment {
```

Comment:
I think we should keep to one name and use `FinalStateCommitment`.
Also, should this be a function or just a field which is calculated in a constructor? `ExecutionResult`  doesn't really change in its lifecycle.
And also, I think this should be on pointer receiver for greater flexibility